File: backend/infrastructure/models/Unit.cs
namespace infrastructure.models;

public class Unit
{
    public int UnitId { get; set; }
    public string Name { get; set; }
    public Status Status { get; set; }
    public UnitType UnitType { get; set; }
}

public enum Status
{
    Armed,
    Disarmed,
    Triggered,
    Closed,
    Opened
}

public enum UnitType
{
    Door,
    Window,
    MotionSensor,
}

File: frontend/src/models/Unit.ts
export interface Unit{
  unitId?: number;
  name: string;
  status: Status;
  unitType: UnitType;
}

export enum UnitType{
  Doors,
  Windows,
  MotionSensor,
}

export enum Status{
  Armed,
  Disarmed,
  Triggered,
  Open,
  Closed,
}
